### PR TITLE
Test support changes

### PR DESCRIPTION
**Pull request by Jesse-Millwood**
This basically adds support for controlling the building variants of libfreertos and configuring it from the test suite as well as controlling the main task. 

This way, a test can provide their own `FreeRTOSConfig.h`, which may have certain #define guards, and pass in `CONFIG_CFLAGS` at the command line invocation from their own Makefile to build a newly named library of the form: `libfreertos_$VARIANT.a `

This functionality is required for things like rhealstone, that need to have specific RTOS configurations in order to test performance of different RTOS features. 

The association of the main isp task with a FreeRTOS handle allows suspending and resuming of the task for tests that need more control over the scheduling.

### CODE REVIEW

**Comment by Jesse-Millwood** [comment]
> squash with previous commit.

I squashed the ones that added the libfreertos variant capability

**Comment by austinharris** [comment]
These failed on CI with:
portASM.S:250: undefined reference to `portasmMSI_HANDLER’

File: Makefile.isp [hunk 1]
```diff
@@ -20,5 +21,11 @@ install: all
 	cp -rf $(RV_DEMO_DIR) $(COMMON_DEMO_DIR) $(DEMO_INSTALL_DIR)
 	cp -rf $(SOURCE_DIR) $(INSTALL_DIR)
 
+install-source:
```
[the remark this answers is not in the document]
Comment (Jesse-Millwood) [reply Makefile.isp]: That would make much more sense

**Comment by Jesse-Millwood** [comment]
> These failed on CI with:
> portASM.S:250: undefined reference to `portasmMSI_HANDLER’

I added a default value for it

**Comment by austinharris** [comment]
Can you please bump the corresponding hope-src branch to point to your new force pushed commit so that CI will run again?

**Comment by arunthomas** [comment]
@austinharris I bumped hope-src for the new changes.

**Pull request merged**